GCC 14 reports a maybe-uninitialized variable

Hi,

When building double-to-string.cc under -O3 -Wall -Wextra with GCC 14.2, a maybe-uninitialized warning is raised:

```
double-conversion/double-to-string.cc: In member function ‘bool double_conversion::DoubleToStringConverter::ToShortestIeeeNumber(double, double_conversion::StringBuilder*, DtoaMode) const’:
double-conversion/double-to-string.cc:197:7: error: ‘decimal_point’ may be used uninitialized [-Werror=maybe-uninitialized]
  197 |   int exponent = decimal_point - 1;
      |       ^~~~~~~~
double-conversion/double-to-string.cc:183:7: note: ‘decimal_point’ was declared here
  183 |   int decimal_point;
      |       ^~~~~~~~~~~~~
```

I think it could be a false positive due to GCC not understanding the logic of the code, but I don't think it would be really harmful here to simply initialize that variable...

--- a/double-conversion/double-to-string.cc
+++ b/double-conversion/double-to-string.cc
@@ -180,7 +180,7 @@ bool DoubleToStringConverter::ToShortestIeeeNumber(
     return HandleSpecialValues(value, result_builder);
   }
 
-  int decimal_point;
+  int decimal_point = 0;
   bool sign;
   const int kDecimalRepCapacity = kBase10MaximalLength + 1;
   char decimal_rep[kDecimalRepCapacity];
@@ -405,6 +405,7 @@ void DoubleToStringConverter::DoubleToAscii(double v,
   if (mode == PRECISION && requested_digits == 0) {
     vector[0] = '\0';
     *length = 0;
+    *point = 0;
     return;
   }
 

